addons/http_routing/models/ir_http.py

```diff
@@ -91,8 +91,13 @@ def slug(value):
     if isinstance(value, models.BaseModel):
         if not value.id:
             raise ValueError("Cannot slug non-existent record %s" % value)
-        # [(id, name)] = value.name_get()
-        identifier, name = value.id, value.display_name
+        # werkzeug 1.0 tries to build() the url during dispatch, at which point
```

Comment:
https://github.com/pallets/werkzeug/pull/1286 specifically here: https://github.com/pallets/werkzeug/blame/048cdfd9b969c0c3a133d7ff43b8ad1ad6a673ec/src/werkzeug/routing.py#L904

An alternative would be to set merge_slashes to False instead, I think that's what I originally did then changed it to this. Possibly because I thought it was an existing feature whose implementation had been changed in 1.0 (and thus didn't want to disable a feature we may have been relying on), but it's actually a new feature as of 1.0, so I don't know that there'd be any issue disabling it.

Reply:
I guess it's ok this way, provided that `merge_slashes` doesn't break anything else, it will spare an extra test for werkzeug 1.0 at another callsite. Through we could add those useful references in the comments, in case we want to change our decision later :-)

I'm also generally wary of type-testing on `env.uid` as it's caused problems [elsewhere](https://github.com/odoo/odoo/commit/276f22c49511ba33b50eda242f39ecbd76e967a9).
I don't suppose we have `auth='none'` routes with a slug, but you might want to change the first branch of the `if` to include `env.uid is None`, just in case.

Reply:
> I guess it's ok this way, provided that merge_slashes doesn't break anything else, it will spare an extra test for werkzeug 1.0 at another callsite. Through we could add those useful references in the comments, in case we want to change our decision later :-)

We probably don't need an extra test, we can just set the attribute explicitly on the rule objects (rather than pass it as a ctor parameter). It should  be a no-op for previous versions.

Reply:
Good idea then, less code and less work in URL processing :+1:

Reply:
Works for me. Any other comments or is it fine to merge after that?

Reply:
Good for me after that, thanks! :-)